File: app/src/main/java/com/sevtinge/cemiuiler/module/home/AlwaysShowMiuiWidget.java
package com.sevtinge.cemiuiler.module.home;

import android.content.ComponentName;

import com.sevtinge.cemiuiler.module.base.BaseHook;

import de.robv.android.xposed.XC_MethodHook;
import de.robv.android.xposed.XC_MethodReplacement;
import de.robv.android.xposed.XposedBridge;
import de.robv.android.xposed.XposedHelpers;

public class AlwaysShowMiuiWidget extends BaseHook {

    Class<?> mWidgetsVerticalAdapter;
    Class<?> mBaseWidgetsVerticalAdapter;
    Class<?> mMIUIAppWidgetInfo;
    Class<?> mMIUIWidgetUtil;

    XC_MethodHook.Unhook hook1;
    XC_MethodHook.Unhook hook2;

    @Override
    public void init() {
        mWidgetsVerticalAdapter = findClassIfExists("com.miui.home.launcher.widget.WidgetsVerticalAdapter");
        mBaseWidgetsVerticalAdapter = findClassIfExists("com.miui.home.launcher.widget.BaseWidgetsVerticalAdapter");
        mMIUIAppWidgetInfo = findClassIfExists("com.miui.home.launcher.widget.MIUIAppWidgetInfo");
        mMIUIWidgetUtil = findClassIfExists("com.miui.home.launcher.MIUIWidgetUtil");


        XposedHelpers.findAndHookMethod(mMIUIAppWidgetInfo, "initMiuiAttribute", ComponentName.class, new MethodHook() {
            @Override
            protected void after(MethodHookParam param) throws Throwable {
                XposedHelpers.setBooleanField(param.thisObject, "isMIUIWidget", false);
            }
        });

        XposedHelpers.findAndHookMethod(mMIUIWidgetUtil, "isMIUIWidgetSupport", XC_MethodReplacement.returnConstant(false));

        /*MethodHook m = new MethodHook() {
            @Override
            protected void before(MethodHookParam param) throws Throwable {
                XposedHelpers.findAndHookMethod(mMIUIAppWidgetInfo, "initMiuiAttribute", ComponentName.class, new MethodHook() {
                    @Override
                    protected void after(MethodHookParam param) throws Throwable {
                        XposedHelpers.setBooleanField(param.thisObject, "isMIUIWidget", false);
                    }
                });

                XposedHelpers.findAndHookMethod(mMIUIWidgetUtil, "isMIUIWidgetSupport", XC_MethodReplacement.returnConstant(false));
            }

            @Override
            protected void after(MethodHookParam param) throws Throwable {
                XposedHelpers.findAndHookMethod(mMIUIAppWidgetInfo, "initMiuiAttribute", ComponentName.class, new MethodHook() {
                    @Override
                    protected void after(MethodHookParam param) throws Throwable {
                        XposedHelpers.setBooleanField(param.thisObject, "isMIUIWidget", true);
                    }
                });

                XposedHelpers.findAndHookMethod(mMIUIWidgetUtil, "isMIUIWidgetSupport", XC_MethodReplacement.returnConstant(true));
            }
        };

        try {
            hookAllMethods(mWidgetsVerticalAdapter, "buildAppWidgetsItems", m);
        } catch (Exception e) {
            hookAllMethods(mBaseWidgetsVerticalAdapter, "buildAppWidgetsItems", m);
        }*/

    }
}

File: app/src/main/java/com/sevtinge/cemiuiler/module/home/widget/AlwaysShowMiuiWidget.kt
package com.sevtinge.cemiuiler.module.home.widget

import com.github.kyuubiran.ezxhelper.ClassUtils.loadClass
import com.github.kyuubiran.ezxhelper.HookFactory.`-Static`.createHook
import com.github.kyuubiran.ezxhelper.finders.MethodFinder.`-Static`.methodFinder
import com.sevtinge.cemiuiler.module.base.BaseHook
import com.sevtinge.cemiuiler.utils.setObjectField
import de.robv.android.xposed.XC_MethodHook

object AlwaysShowMiuiWidget : BaseHook() {
    override fun init() {
        var hook1: XC_MethodHook.Unhook? = null
        var hook2: XC_MethodHook.Unhook? = null
        try {
            loadClass("com.miui.home.launcher.widget.WidgetsVerticalAdapter").methodFinder().first {
                name == "buildAppWidgetsItems"
            }
        } catch (e: Exception) {
            loadClass("com.miui.home.launcher.widget.BaseWidgetsVerticalAdapter").methodFinder().first {
                name == "buildAppWidgetsItems"
            }
        }.createHook {
            before {
                hook1 = loadClass("com.miui.home.launcher.widget.MIUIAppWidgetInfo").methodFinder()
                    .first {
                        name == "initMiuiAttribute" && parameterCount == 1
                    }.createHook {
                        after {
                            it.thisObject.setObjectField("isMIUIWidget", false)
                        }
                    }
                hook2 = loadClass("com.miui.home.launcher.MIUIWidgetUtil").methodFinder().first {
                    name == "isMIUIWidgetSupport"
                }.createHook {
                    after {
                        it.result = false
                    }
                }
            }
            after {
                hook1?.unhook()
                hook2?.unhook()
            }
        }
    }
}
